tests/foreman/api/test_computeprofile.py

```diff
@@ -0,0 +1,84 @@
+# -*- encoding: utf-8 -*-
+"""Unit tests for the Compute Profile feature."""
+from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo.datafactory import invalid_values_list, valid_data_list
+from robottelo.decorators import tier1
+from robottelo.test import APITestCase
+
+
+class ComputeProfileTestCase(APITestCase):
+    """Tests for compute profiles."""
+
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create new Compute Profile using different names
+
+        @Feature: Compute Profile - Positive Create
+
+        @Assert: Compute Profile is created
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                profile = entities.ComputeProfile(name=name).create()
+                self.assertEqual(name, profile.name)
+
+    @tier1
+    def test_negative_create(self):
+        """Attempt to create Compute Profile using invalid names only
+
+        @Feature: Compute Profile - Negative Create
+
+        @Assert: Compute Profile is not created
+        """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.ComputeProfile(name=name).create()
+
+    @tier1
+    def test_positive_update_name(self):
+        """Update selected Compute Profile entity using proper names
+
+        @Feature: Compute Profile - Positive Update
```

Comment:
I like your idea and actually wanted to have implementation like that in the past, but decided to follow overall style. Don't mind to discuss the idea in future